### PR TITLE
Permission and Group Refactor

### PR DESCRIPTION
**Pull request by ArmaanT**
This PR:
* Refactors PennAffiliation to use Django's built-in groups
* Refactors ProductPermission to use Django's built-in permissions

### CODE REVIEW

**Comment by ArmaanT** [comment]
[ch101]

File: accounts/backends.py [hunk 1]
```diff
@@ -49,11 +50,12 @@ def authenticate(self, request, remote_user, shibboleth_attributes):
...
                 setattr(user, key, value)
 
-        # Update affiliations with every log in
-        user.affiliation.clear()
+        # Update groups with every log in
+        user.groups.clear()
         for affiliation_name in shibboleth_attributes["affiliation"]:
-            affiliation, _ = PennAffiliation.objects.get_or_create(name=affiliation_name)
-            user.affiliation.add(affiliation)
+            if affiliation_name:  # Some users don't have any affiliation somehow ¯\_(ツ)_/¯
```
Comment (davish) [inline accounts/backends.py]: would explicitly say `if affiliation_name is not None` –– don't like relying on falsiness in Python.

File: accounts/migrations/0002_auto_20200213_1711.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            user.save()
+
+
+def copy_permissions(apps, schema_editor):
+    with transaction.atomic():
+        User = apps.get_model("accounts", "User")
+        Permission = apps.get_model("auth", "Permission")
+        for user in User.objects.all():
+            for product_permission in user.product_permission.all():
+                group = Permission.objects.get(codename=product_permission.id)
```
Comment (davish) [inline accounts/migrations/0002_auto_20200213_1711.py]: No `get_or_create` on this one?

File: accounts/migrations/0002_auto_20200213_1711.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        migrations.DeleteModel(name="PennAffiliation",),
+        migrations.AlterModelOptions(
+            name="user",
+            options={
+                "permissions": (
+                    ("cfa_admin", "Admin for Common Funding Application"),
+                    ("clubs_admin", "Admin for Penn Clubs"),
+                    ("courses_admin", "Admin for Penn Courses"),
+                    ("pcr_admin", "Admin for Penn Course Review"),
+                    ("studentlife_admin", "Admin for Student Life"),
```
Comment (davish) [inline accounts/migrations/0002_auto_20200213_1711.py]: is there a way to add these dynamically or will it require a code change every time?

File: accounts/backends.py (hunk 1 above)
Comment (ArmaanT) [reply accounts/backends.py]: Makes sense

File: accounts/migrations/0002_auto_20200213_1711.py (hunk 3 above)
Comment (ArmaanT) [reply accounts/migrations/0002_auto_20200213_1711.py]: We definitely could do it dynamically, but django seems to[ suggest doing it this way](https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#custom-permissions). Honestly I don't know why or what the tradeoffs are though.

**Pull request merged**